spec/dummy/config/environments/test.rb

```diff
@@ -21,6 +21,15 @@
 
   config.view_component_storybook.show_stories = true
 
+  config.view_component_storybook.stories_title_generator = lambda { |stories|
+    # Test a custom story title generator
+    if stories.stories_name == "demo/heading_component"
+      return stories.stories_name.delete_prefix('demo/').humanize.titlecase
+    end
+
+    stories.stories_name.humanize.titlecase
+  }
```

Comment:
Yes, I did think this also, but we can't do this unfortunately as the config is gets picked up by the classes on load - changing the config later has no affect (hence my unpleasant attempt at getting some coverage on this)

I suppose it could be argued we're testing effectively Rails here, but it just feels wrong to add this functionality here with no coverage.. I'm stuck :-/

Reply:
I wonder if you can set the config in an around block and declare the class in the test with something like `Class.new(ViewComponent::Storybook::Stories)` and `stub_const` something like this: https://bloggie.io/@kinopyo/rspec-define-inline-class

Reply:
That's a great idea - I never thought of that, sounds like it's what's needed - I'll get on it, thanks!

Reply:
@jonspalmer I spent a bit of time on this earlier, I'm not sure it's moving towards a better solution for a few reasons:

1. We can't really easily redefine the config and pick it up, setting it on the class is the most straightforward approach I think
2. The dynamic class `name` isn't available when it inherits `ViewComponent::Storybook::Stories` (causing issues when trying to build the `stories_name`), the cleanest way around this that I can see right now is to stub a response for this this, but this feels weird
3. ActiveSupport descendant tracking records the dynamic class in the list of descendants, so this needs to be disabled somehow (I've stubbed it out for now)

Not sure if it's worth continuing on this as it still doesn't feel quite right to me, but maybe it's ok?

FWIW, here's what I have currently:

```ruby
    context 'with a custom story title generator defined' do
      let(:stories_class_name) { 'Demo::ButtonComponentStories' }

      before do
        # Creating the class dynamically leaves `name` nil
        allow(ViewComponent::Storybook::Stories).to receive(:name).and_return(stories_class_name)
        # Descendant tracking appends our dynamic class to the list of
        # descendants which (logically) causes failures on the .all example
        # below
        allow(ActiveSupport::DescendantsTracker).to receive(:store_inherited)
        stub_const(stories_class_name, Class.new(ViewComponent::Storybook::Stories))
      end

      around do |example|
        original_generator = ViewComponent::Storybook.stories_title_generator
        ViewComponent::Storybook.stories_title_generator = lambda { |stories|
          stories.stories_name.delete_prefix('demo/').humanize.titlecase
        }
        example.run
        ViewComponent::Storybook.stories_title_generator = original_generator
      end

      it "converts Stories" do
        expect(Demo::ButtonComponentStories.to_csf_params).to eq(
          title: "Button Component",
          stories: []
        )
      end
    end
```

Reply:
That does feel unwieldily. Can we fix up this PR, revert the title changes we talked about and then leave some failing spec or the like for us to investigate? I'd like to be able to checkout this change and play around with a few options.

Reply:
Digging in a bit more I think this example is fine. To you points 1. & 3.  seem fine. We could stub the name as in 2. but I wonder if we just go with some static title in the lambda or some other property of the `stories` param just to prove it works even though its not a very practical real world example.

Thoughts?